astropy/table/pprint.py

```diff
@@ -546,7 +546,10 @@ def format_col_str(idx):
                 except ValueError:
                     raise ValueError(
                         f'Unable to parse format string "{col_format}" for '
-                        f'entry "{col[idx]}" in column "{col.info.name}"'
+                        f'entry "{col[idx]}" in column "{col.info.name}" '
+                        f'with datatype "{col.info.dtype}". '
+                        "See https://docs.python.org/3/library/string.html#format-specification-mini-language "
```

Comment:
I'd suggest to link to the astropy table docs on the format specification https://docs.astropy.org/en/stable/table/construct_table.html#format-specifier. This documents all the options including the same link to the format mini-language.